### PR TITLE
Refactor: Migrate Bootstrap Table to DataGrid Issue #173

### PR DESCRIPTION
**Pull request by annovo**
Pull-Request for `paretOS`

## Description
- Added `@mui/x-data-grid` package
- Replaced table with DataGrid
- Added StyledDataGrid for custom DataGrid styling

## Relates to
- [Refactor: Migrate one-off Bootstrap Components Table to Material UI](https://github.com/mikhael28/paretOS/issues/173)

## Reviewers
- @mikhael28 
- @jayeclark 

### Notes 
- The selected/hover style is currently aligned with `theme.palette.primary` instead of the original orange styling.
- The rank is displayed for all people instead of "-" for equal ranking. The primary reason for it is confusing results when sorting. Like this: 
Screen Shot 2022-04-18 at 4 23 53 PM

### Screenshots / other info

Screen Shot 2022-04-18 at 6 16 05 PM

### CODE REVIEW

**Comment by mikhael28** [comment]
@annovo thank you for sending this in! You did a great job in recreating the look and feel of the current leaderboard, and did it very quickly.

The thing I am worried about, as the size of the OS (in terms of functionality, and total amount of code) becomes bigger, is performance and bundle size. The one nice thing about the `react-bootstrap` table, is that it is only 15kb
Screen Shot 2022-04-19 at 8 14 26 AM

However, looking at my favorite site bundlephobia, I can see that the `mui/x-data-grid` community edition is coming in at 450kb, 120kb gzipped. So, it's a 10x increase in size for maybe 1.5x improvement in functionality.
Screen Shot 2022-04-19 at 8 12 52 AM

For context, the 'main' application bundle that is required for download before true functionality can be activated is 442kb.
Screen Shot 2022-04-19 at 8 20 32 AM
 Here is the screenshot of the JavaScript bundles (we are using bundle splitting/lazy loading to dynamically download them at run-time, to reduce the initial bundle size, so that the download length is less and our time to paint is faster! More details on web vitals: https://web.dev/vitals/

Before we can agree to move forward with this package, I would like you to look at a few things.

1. Can you run `yarn build` and upload a screenshot of the impact on the bundle size, compared to the one I posted above. You can also use lazy loading/bundle splitting on the Leaderboard component to try and decrease the impact, refer to `Routes.js` to see how this is done.
2. Can you research what other react table components are out there? Especially, are there any that are razor focused on performance and bundle size? This is a good article to start with: https://blog.logrocket.com/the-top-react-table-libraries-to-use-in-2021/#:~:text=React%20Table%20is%20a%20lightweight,14kb)%2C%20composable%20and%20extensible.
3. Based on your research, can you present the three best candidates for us?

Again, thank you for doing the initial work here - the mui/x-data-grid professional package certainly looks compelling - in their demo on the website, it easily renders 3 million cells - crazy! I think for us, that might be a little overkill :)

I took some time to write this comment out, I hope it will be a helpful experience to learn to focus on bundle size and performance!

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "@emotion/styled": "11.8.1",
     "@mui/material": "5.5.0",
     "@mui/styles": "5.5.0",
+    "@mui/x-data-grid": "^5.9.0",
```
Comment (mikhael28) [inline package.json:14]: Refer to my comment in the PR!

**Review by mikhael28 — CHANGES_REQUESTED** [review]

**Comment by mikhael28** [comment]
You can run also run `yarn run analyze`, if you installed `source-map-explorer`, to create a visual matrix of the bundle size as well!

**Comment by jayeclark** [comment]
This one is on me, for suggesting using DataGrid without thinking to look at the size of the bundle. (I was thinking ahead to a time when we'll need to display hundreds or thousands of accounts compared to how the component is currently used.)

The basic Table component probably does what's needed right now perfectly fine and is only slightly larger than bootstrap. 
Screen Shot 2022-04-19 at 12 50 09 PM

**Comment by mikhael28** [comment]
The basic Table component would work great for right now! @annovo does that sound good?

**Comment by annovo** [comment]
I used lazy loading of Board component and here what I got
Screen Shot 2022-04-19 at 1 27 47 PM

Surely the third bundle size is due to DataGrid component
Screen Shot 2022-04-19 at 1 32 58 PM

I agree that the Table MUI component should be fine.

For example, react-table (55k/15.4k) provides a lot of hooks and definitely can be useful in some cases. However, with the existing functionality this much is unnecessary. Also, it is possible to write custom hooks from the existing sort/filter functions for the table. 
react-virtulized(118k)/react-window(24k) can be used in the future for large data. Here the [example]( https://codesandbox.io/s/je8fbn?file=/demo.tsx) of using it with MUI.

**Comment by annovo** [comment]
> The basic Table component would work great for right now! @annovo does that sound good?

whoops, sorry, haven't noticed. Yes, I can do the basic Table. Should I make a new PR?

**Comment by jayeclark** [comment]
@annovo I think it would be fine to just push a new commit to this branch (or even force-push a new one rewriting the last two.) I don't know how you feel about your merged-vs-closed-PR ratio, but personally I like to keep mine high. 😄 And this wasn't an error on your part, it was an oversight on mine!

**Comment by mikhael28** [comment]
@annovo thank you for posting the bundle size on the x-data-grid - yes, let's go ahead and integrate the basic Table component from MUI. Let's keep it on this PR :) just make a few new commits and push up the changes!

**Comment by annovo** [comment]
New table looks pretty much the same
Screen Shot 2022-04-21 at 9 43 04 AM

**Review by mikhael28 — APPROVED** [review]
Great work! The Leaderboard looks great, we are one step closer to ridding ourselves of the accursed `react-bootstrap`!

**Pull request merged**